hw/ip/otbn/dv/otbnsim/sim/dmem.py

```diff
@@ -219,16 +224,17 @@ def _commit_store(self, item: TraceDmemStore) -> None:
                 wr_data = (item.value >> (i * 32)) & mask
                 self.data[(item.addr // 4) + i] = wr_data
 
-            return
         else:
+            assert 0 <= item.value <= (1 << 32) - 1
             self.data[item.addr // 4] = item.value
 
-        assert 0 <= item.value <= (1 << 32) - 1
-
     def commit(self) -> None:
         for item in self.trace:
             self._commit_store(item)
         self.trace = []
 
     def abort(self) -> None:
         self.trace = []
+
+    def EmptyDmem(self) -> None:
+        self.data = [None] * self.num_words
```

Comment:
Probably easier to do `self.data = [None] * len(self.data)`, which avoids needing to define `self.num_words`.